subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -105,28 +136,22 @@ int fota_download_start(char *host, char *file)
 	}
 
 	/* Verify that a download is not already ongoing */
-	if (dfu.fd != -1) {
+	if (dlc.fd != -1) {
 		return -EALREADY;
 	}
 
-	int err = flash_img_init(&flash_img);
-
-	if (err != 0) {
-		LOG_ERR("flash_img_init error %d", err);
-		return err;
-	}
-
-	err = download_client_connect(&dfu, host, &config);
+	err = download_client_connect(&dlc, host, &config);
 
 	if (err != 0) {
 		LOG_ERR("download_client_connect error %d", err);
 		return err;
 	}
 
-	err = download_client_start(&dfu, file, 0);
+
+	err = download_client_start(&dlc, file, 0);
 	if (err != 0) {
 		LOG_ERR("download_client_start error %d", err);
```

Comment:
All errors in `download_client_start()` are already logged in that function, so you can remove the log here, it's redundant.